controllers/packagebundlecontroller_controller.go

```diff
@@ -147,12 +146,22 @@ func (r *PackageBundleControllerReconciler) Reconcile(ctx context.Context, req c
 	}
 
 	if !r.bundleManager.IsBundleKnown(ctx, bundles.Items, latestBundle) {
+		r.Log.V(6).Info("New latest bundle", "bundle", latestBundle.Name)
 		err = r.Client.Create(ctx, latestBundle)
 		if err != nil {
 			return result, fmt.Errorf("creating new package bundle: %s", err)
 		}
 	}
 
+	if pbc.Spec.ActiveBundle == "" {
```

Comment:
I have concerns about the implications of this flow. It's not clear to me that when there is no active bundle, that the right choice is to activate the newest bundle.

Outside of first-run conditions, this should never happen, but it if does, it could be disastrous to upgrade someone's bundle out from under them. There's no logic here to check for a first-run condition (which is why I like having a bundle from the start, it eliminates the need for a first-run check, and makes any case where activebundle is blank an error state).

I think I'd feel a lot better if there were something else in the spec or status of a new controller that indicated, through it's presence (rather than through its absence) that this was a first-run situation.

Reply:
Let me think what we can do here.